### DISCUSSION TITLE
'until'-Loop

### DISCUSSION THREAD

**Discussion by PhilippElhaus** [dotnet/csharplang#7313] [Language Ideas]
Add a new control structure that complements the existing `while` loop:

the `until` loop.

The `until` loop is basically a negated `while`-loop:

`while(!condition) { ... }` equals
`until(condition) { ... }`

The loop executes 'until' a certain condition is met. This idea is inspired from BASH Shell-Scripting where this is already a thing.

**Comment by TahirAhmadov** [comment]
Too much risk WRT to keyword and collision with identifiers, to save one `!`

**Comment by huoyaoyuan** [comment]
See also #157 #560 .
I remember there's a proposal about allowing putting `!` outside braces: `if !(complex condition)`, which can save a pair of braces.

**Comment by The-Futurist** [comment]
Some other languages offer both a `while` and an `until` variant. The `until` is similar to the `do {...} while` in C#. It seems to me that the requested change adds little, given that we can already express what's desired.
